packages/hdwallet-tallyho/src/tallyho.ts

```diff
@@ -62,6 +63,53 @@ export class TallyHoHDWalletInfo implements core.HDWalletInfo, core.ETHWalletInf
     return chainId === 1;
   }
 
+  private async detectTallyProvider(): Promise<TallyHoEthereumProvider | null> {
```

Comment:
Note: We might get away with a lighter implementation since the checks for being in the context of a Tally provider should be completed, but IMO doesn't hurt to err on the side of safety and keep the full detection implementation